tests/test_gf_unpack2.F90

```diff
@@ -3,255 +3,128 @@
 ! This program tests the gf_unpack2.f file.
 !
 ! Andrew King 6/23/23
-
- program test_gf_unpack2
-    
-    implicit none
-
-    character, dimension(269) :: fgrib
-    integer :: k1, k2, k3, k4, k5, k6, k7, k8, k9, k10
-    integer :: n1, n2, n3, nan, fgrib_len
-    integer :: iofst, lencsec2, lensec, ierr, i
-    integer :: isecnum, numlocal, numfields
-    integer :: listsec0(3), listsec1(13), maxvals(7)
-    character(len=1),pointer,dimension(:) :: csec2
-
-    interface
-        subroutine gf_unpack2(cgrib, lcgrib, iofst, lencsec2, csec2, ierr)
-        character(len = 1), intent(in) :: cgrib(lcgrib)
-        integer, intent(in) :: lcgrib
-        integer, intent(inout) :: iofst
-        integer, intent(out) :: lencsec2
-        integer, intent(out) :: ierr
-        character(len = 1), pointer, dimension(:) :: csec2
-        end subroutine gf_unpack2
-    end interface
-
-    ! To initialize large parameters in typical grib2 file.       
-    k1=6371200                   ! Radius of the earth
-    k2=1073                      ! Nx for lambert 5km
-    k3=689                       ! Ny for lambert 5km
-    k4=2019199                   ! La1 - latitude of first grid point 
-    k5=2384459                   ! Lo1 - longitude of first grid point
-    k6=25000000                  ! Latitude where grid spacing is defined
-    k7=265000000                 ! Orientation longitude
-    k8=2539703                   ! X & Y direction grid length
-    k9=-90000000                 ! Latitude of pole 
-    k10=739297                   ! Number of grid Nx*Ny
-    Nan=9999                     ! Missing value
-    fgrib_len=269                ! Grib message length
-
-    ! Large numbers in message initials
-    n1=229
-    n2=254
-    n3=255
-
-    ! The grib message for a typical grib2 file in fortran.
-    fgrib(:)=(/                                                    &
-    ! section 0
-    & "G", "R", "I", "B", char(0), char(0),                        &
-    & char(0), char(2), char(0), char(0), char(0), char(0),    &
-    & char(0), char(0), char(1), char(13),                       &
-        ! section 1
-    & char(0), char(0), char(0), char(21), char(1), char(0),   &
-    & char(8), char(0), char(0), char(1), char(0), char(1),    &
-    & char(7), char(n1), char(11), char(15), char(10),          &
-    & char(10), char(10), char(0), char(1),                      &
-        ! section 2
-    & char(0), char(0), char(0), char(11), char(2),             &
-    & char(1), "H", "K", "T", "A", "R",                             &
-        ! section 3
-    & char(0), char(0), char(0), char(81), char(3), char(0),   &
-    & char(0), char(0), char(0), char(k10), char(0), char(0),  &
-    & char(0), char(30), char(1), char(0), char(k1), char(k1), &
```

Comment:
OK, this is incorrect in several places. THe problem is that the char() function returns a character*1, which is 8 bytes. So no value greater than 255 makes sense as the argument.